lib/BloqadeGates/src/types.jl

```diff
@@ -0,0 +1,41 @@
+struct RydbergPulse{D} <: AbstractBlock{D}
+    rydberg_hamiltonian::Union{RydbergHamiltonian, RydbergHamiltonian3}
+    t_start::Float64
+    t_end::Float64
+    backend::Union{Type{KrylovEvolution}, Type{SchrodingerProblem}}
```

Comment:
put this in the type parameter?

Reply:
Do you mean by `t_start` and `t_end`?

Reply:
I mean `backend`, it is a type right? why not just put them in type parameters